pkg/cmd/cli/cmd/request_project.go

```diff
@@ -0,0 +1,74 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+)
+
+type NewProjectOptions struct {
+	ProjectName string
+	DisplayName string
+	Description string
+
+	Client client.Interface
+}
+
+func NewCmdRequestProject(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &NewProjectOptions{}
+
+	cmd := &cobra.Command{
+		Use:   name + " <project-name>",
+		Short: "request a new project",
+		Long:  `provisions a new project with you as the project admin`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if !options.complete(cmd) {
+				return
+			}
+
+			var err error
+			if options.Client, _, err = f.Clients(); err != nil {
+				glog.Fatalf("Error getting client: %v", err)
+			}
+			if err := options.Run(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+	cmd.SetOutput(out)
+
+	cmd.Flags().StringVar(&options.DisplayName, "display-name", "", "project display name")
```

Comment:
separate pull perhaps?  project.go needs some refactoring before it's able to handle easy delegation.